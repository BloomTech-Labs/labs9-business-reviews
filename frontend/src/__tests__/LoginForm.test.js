import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import LoginFormComponent from '../components/LoginForm';

describe('<LoginComponent />', () => {
  it('renders', () => {
    shallow(<LoginFormComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<LoginFormComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain 1 h4 with text "Dont have an account?"', () => {
    let app = shallow(<LoginFormComponent />);
    expect(app.find("h4").length).toBe(1);
    expect(app.find("h4").text()).toMatch("Don't have an account?");
  });
  it('should contain one Link', () => {
    let app = shallow(<LoginFormComponent />);
    expect(app.find("Link").length).toBe(1);
  });
  it('should contain 2 label elements', () => {
    let app = shallow(<LoginFormComponent />);
    expect(app.find("label").length).toBe(2);
  });
  it('should contain one input element', () => {
    let app = shallow(<LoginFormComponent />);
    expect(app.find("input").length).toBe(1);
  });
});
