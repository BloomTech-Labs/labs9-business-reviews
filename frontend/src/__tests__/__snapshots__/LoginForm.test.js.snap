// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginComponent /> matches snapshot 1`] = `
<styled.div>
  <styled.form
    onSubmit={[Function]}
  >
    <styled.h1>
      Login to your account
    </styled.h1>
    <h4
      className="Login__text"
    >
      Don't have an account?
       
      <Link
        className="Login__text--link"
        replace={false}
        to="/register"
      >
        Register here!
      </Link>
    </h4>
    <label
      htmlFor="email"
    >
       Email
    </label>
    <styled.input
      name="email"
      onChange={[Function]}
      type="email"
    />
    <label
      htmlFor="password"
    >
       Password
    </label>
    <styled.input
      name="password"
      onChange={[Function]}
      type="password"
    />
    <input
      type="submit"
      value="Login"
    />
  </styled.form>
</styled.div>
`;
